File: laba_2/task_3/task_3.java
import java.util.Scanner;

public class Main {
    
    public static int sumOfDigits(int num) {
        int sum = 0;
        while (num > 0) {
            sum += num % 10;
            num /= 10;
        }
        return sum;
    }
   
    public static int safeInput(Scanner scanner, String message) {
        while (true) {
            System.out.print(message);
            String line = scanner.nextLine();
            
            try {
                return Integer.parseInt(line.trim());
            } catch (NumberFormatException e) {
                System.out.println("Ошибка! Нужно ввести ЧИСЛО. Попробуйте ещё раз!");
            }
        }
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Ввод N с проверкой на положительное число
        int N;
        do {
            N = safeInput(scanner, "Введите N натуральных чисел: ");
            if (N <= 0) {
                System.out.println("Ошибка! N должно быть больше 0. Попробуйте ещё раз!");
            }
        } while (N <= 0);
        
        int count = 0;
        
        for (int i = 0; i < N; i++) {
            int num = safeInput(scanner, "Введите число " + (i + 1) + ": ");
            if (sumOfDigits(num) > 10) {
                count++;
            }
        }
        
        System.out.println("Количество чисел сумма которых больше 10 равна " + count);
        scanner.close();
    }
}

File: laba_2/task_3/task_3.kt
import java.util.Scanner

fun sumOfDigits(num: Int): Int {
    var n = if (num < 0) -num else num  // обрабатываем отрицательные числа
    var sum = 0
    while (n > 0) {
        sum += n % 10
        n /= 10
    }
    return sum
}

fun main() {
    val scanner = Scanner(System.`in`)
    
    var N: Int
    do {
        print("Введите N натуральных чисел: ")
        while (!scanner.hasNextInt()) {
            println("Ошибка! Нужно ввести ЧИСЛО. Попробуйте ещё раз!")
            scanner.next()  // очищаем неправильный ввод
            print("Введите N натуральных чисел: ")
        }
        N = scanner.nextInt()
        if (N <= 0) {
            println("Ошибка! N должно быть больше 0. Попробуйте ещё раз!")
        }
    } while (N <= 0)
    
    var count = 0
    for (i in 1..N) {
        print("Введите число $i: ")
        while (!scanner.hasNextInt()) {
            println("Ошибка! Нужно ввести ЧИСЛО. Попробуйте ещё раз!")
            scanner.next()
            print("Введите число $i: ")
        }
        val num = scanner.nextInt()
        if (sumOfDigits(num) > 10) {
            count++
        }
    }
    
    println("Количество чисел, сумма цифр которых больше 10, равна $count")
}